### PR TITLE
add location override example

### LINKED ISSUE #501: add examples
# TL;DR

- custom dimension example
- location change example
- event example
- cross-domain example

### PR DESCRIPTION
**Pull request by shoota**
# TL;DR

Closes #501

### CODE REVIEW

File: examples/simple/page.pug [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+block footer_content
+  script.
+    !function(e,t,s,n,a,r,c,i){e[a]=e[a]||function(){(e[a].q=e[a].q||[]).push(arguments)},c=t.createElement(s),i=t.getElementsByTagName(s)[0],
+    c.async=1,c.src=n,c.charset=r,c.setAttribute("data-ud-namespace",a),i.parentNode.insertBefore(c,i)}
+    (window,document,"script","https://cdn.userdive.com/agent.js","_ud","UTF-8");
+    _ud("create", 'af57h6gb', 'auto');
+    _ud('send', 'pageview');
+    document.getElementById('page')
+    .addEventListener('click', function () {
+      _ud('send', 'pageview', { page: location.origin+location.pathname+'/dummy'+location.search });
```
Comment (9renpoto) [inline examples/simple/page.pug]: v0.15 does not work yet;
https://github.com/userdive/agent.js/blob/v0.15.0/packages/agent/src/agent.ts#L46

Comment (9renpoto) [reply examples/simple/page.pug]: why not use `_ud('send', 'pageview', pathname)` ?

Comment (9renpoto) [reply examples/simple/page.pug]: based api syntax https://developers.google.com/analytics/devguides/collection/analyticsjs/pages

Comment (shoota) [reply examples/simple/page.pug]: Yes, i know that agent has this interface and `pathname2href` in core.ts work for too.
But this example describe the `page` field can override like [this](https://developers.google.com/analytics/devguides/collection/analyticsjs/pages#modifying_page_urls).

File: examples/test/tracker/page.test.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  assert.equal(envLogger.requests.length, 1)
+  const originalRecord = envLogger.requests[0]
+  assert.equal(query(originalRecord).l, l)
+
+  await t
+    .click(Selector('#page'))
+    .wait(2000)
+  assert.equal(envLogger.requests.length, 2)
+  const overrideRecord = envLogger.requests[1]
+  assert.equal(query(overrideRecord).l, override)
```
Comment (shoota) [inline examples/test/tracker/page.test.ts]: location override test

**Pull request merged**